libmogwai-schedule/tests/scheduler.c

```diff
@@ -909,6 +909,354 @@ test_scheduler_scheduling_metered_connection (Fixture       *fixture,
+      { { tariff_unmetered, }, 2018, 2, 3, 17, 0, 0, NULL, TRUE,
+        { tariff1, }, 2018, 2, 3, 17, 0, 0, NULL, TRUE },
+      { { tariff_unmetered, }, 2018, 2, 4, 1, 30, 0, NULL, TRUE,
+        { tariff1, }, 2018, 2, 4, 1, 30, 0, NULL, FALSE },
+      /* Transition from an unmetered period and a metered period in one
+       * tariff, to a metered period in one and an unmetered period in another.
+       * Since the scheduler requires *all* connections to be safe in order to
+       * start a download, this leads to very limited active downloads. */
+      { { tariff1, tariff2, }, 2018, 2, 4, 0, 30, 0, NULL, FALSE,
+        { tariff1, tariff2, }, 2018, 2, 4, 1, 15, 0, NULL, FALSE },
+      { { tariff1, tariff2, }, 2018, 2, 4, 1, 45, 0, NULL, FALSE,
+        { tariff1, tariff2, }, 2018, 2, 4, 1, 59, 0, NULL, FALSE },
+      { { tariff1, tariff2, }, 2018, 2, 4, 2, 0, 0, NULL, TRUE,
+        { tariff1, tariff2, }, 2018, 2, 4, 2, 15, 0, NULL, TRUE },
+      /* Timezone change in metered and unmetered periods. The underlying UTC
+       * time may change at the same time, to either give a timezone change only
+       * (where the underlying UTC time is the same) or a timezone and time
+       * change such that the resolved wall clock time stays the same */
+      /* Change timezone only: */
+      { { tariff1, }, 2018, 2, 3, 15, 30, 0, "Europe/London" /* UTC+0 */, TRUE,
+        { tariff1, }, 2018, 2, 3, 15, 30, 0, "Australia/Brisbane" /* UTC+10 */, TRUE },
+      /* Keep wall clock time unchanged: */
+      { { tariff1, }, 2018, 2, 3, 1, 30, 0, "Australia/Brisbane" /* UTC+10 */, FALSE,
+        { tariff1, }, 2018, 2, 3, 11, 30, 0, "Europe/London" /* UTC+0 */, TRUE },
+
+      { { tariff1, }, 2018, 2, 3, 16, 30, 0, "Europe/London" /* UTC+0 */, TRUE,
+        { tariff1, }, 2018, 2, 3, 16, 30, 0, "Europe/Berlin" /* UTC+1 */, TRUE },
+      { { tariff1, }, 2018, 2, 3, 15, 30, 0, "Europe/Berlin" /* UTC+1 */, TRUE,
+        { tariff1, }, 2018, 2, 3, 16, 30, 0, "Europe/London" /* UTC+0 */, TRUE },
+
+      { { tariff1, }, 2018, 2, 3, 1, 30, 0, "Europe/London" /* UTC+0 */, FALSE,
+        { tariff1, }, 2018, 2, 3, 1, 30, 0, "Australia/Brisbane" /* UTC+10 */, FALSE },
+      { { tariff1, }, 2018, 2, 2, 15, 30, 0, "Australia/Brisbane" /* UTC+10 */, TRUE,
+        { tariff1, }, 2018, 2, 3, 1, 30, 0, "Europe/London" /* UTC+0 */, FALSE },
+
+      { { tariff1, }, 2018, 2, 3, 1, 30, 0, "Europe/London", FALSE,
+        { tariff1, }, 2018, 2, 3, 1, 30, 0, "Europe/Isle_of_Man", FALSE },
+      /* Test different times in (and out of) a tariff which doesn’t repeat. */
+      { { tariff3, }, 2017, 12, 30, 0, 0, 0, NULL, TRUE,
+        { tariff3, }, 2018, 1, 1, 1, 30, 0, NULL, FALSE },
+      { { tariff3, }, 2018, 1, 1, 0, 0, 0, NULL, FALSE,
+        { tariff3, }, 2018, 1, 2, 0, 0, 0, NULL, TRUE },
+    };
+
+  for (gsize i = 0; i < G_N_ELEMENTS (transitions); i++)
+    {
+      /* Set up the datess/times. The date/time figures in the test vector are
```

Comment:
![datess-datesessssss](https://user-images.githubusercontent.com/86760/41669638-2a9bbe4a-74aa-11e8-9c39-7de0260445a1.jpg)

Reply:
:laughing: This makes me want to keep the typo in now.

Reply:
Fine by me

Reply:
Fixup pushed.